File: angularapp/src/app/services/order.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Orders } from '../models/orders.model';
import { Observable } from 'rxjs';
import { API_URL } from '../app.constant';

@Injectable({
  providedIn: 'root'
})
export class OrderService {

  constructor(private client:HttpClient) { }
  
  placeOrder(order:Orders):Observable<Orders>{
    return this.client.post(`${API_URL}/orders`,order);
  }

  getAllOrders():Observable<any>{
    return this.client.get(`${API_URL}/orders/all`);
  }

  //Pagination 
  getAllPaginatedOrders(page : number , size: number):Observable<any>{
    return this.client.get(`${API_URL}/orders?page=${page}&size=${size}`);
  }

  getOrdersById(orderId:number):Observable<Orders>{
    return this.client.get(`${API_URL}/orders/${orderId}`)
  }
  
  getAllOrdersByUserId(userId:number):Observable<any>{
    return this.client.get(`${API_URL}/orders/user/${userId}`)
  }

  //Pagination
  getAllPaginatedOrdersByUserId(userId:number,page : number , size: number):Observable<any>{
    return this.client.get(`${API_URL}/orders/user/${userId}?page=${page}&size=${size}`)
  }

  updateOrder(orderId:number,order:Orders):Observable<Orders>{
    return this.client.put(`${API_URL}/orders/${orderId}`,order)
  }

  deleteOrder(orderId: number):Observable<void>{
    return this.client.delete<void>(`${API_URL}/orders/${orderId}`)
  }
}

File: springapp/src/main/java/com/examly/springapp/service/OrderService.java
package com.examly.springapp.service;

import java.util.List;
import java.util.Optional;

import org.springframework.data.domain.Page;

import com.examly.springapp.model.Orders;

public interface OrderService {

    Orders addOrder(Orders order);

    Optional<Orders> getOrderById(int orderId);

    // List<Orders> getOrdersByUserId(long userId);
    Page<Orders> getPaginatedOrdersByUserId(long userId , int page , int size);

    List<Orders> getAllOrders();
    Page<Orders> getAllPaginatedOrders(int page , int size);

    Orders updateOrder(Orders order, int orderId);

    boolean deleteOrder(int orderId);

}
